sdk/src/Services/S3/Generated/Model/Internal/MarshallTransformations/ListBucketMetricsConfigurationsRequestMarshaller.cs

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ * 
+ *  http://aws.amazon.com/apache2.0
+ * 
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/*
+ * Do not modify this file. This file is generated from the s3-2006-03-01.normal.json service model.
+ */
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Xml.Serialization;
+
+using Amazon.S3.Model;
+using Amazon.Runtime;
+using Amazon.Runtime.Internal;
+using Amazon.Runtime.Internal.Transform;
+using Amazon.Runtime.Internal.Util;
+using System.Xml;
+
+#pragma warning disable CS0612,CS0618
+namespace Amazon.S3.Model.Internal.MarshallTransformations
+{
+    /// <summary>
+    /// ListBucketMetricsConfigurations Request Marshaller
+    /// </summary>       
+    public partial class ListBucketMetricsConfigurationsRequestMarshaller : IMarshaller<IRequest, ListBucketMetricsConfigurationsRequest> , IMarshaller<IRequest,AmazonWebServiceRequest>
+    {
+        /// <summary>
+        /// Marshaller the request object to the HTTP request.
+        /// </summary>  
+        /// <param name="input"></param>
+        /// <returns></returns>
+        public IRequest Marshall(AmazonWebServiceRequest input)
+        {
+            return this.Marshall((ListBucketMetricsConfigurationsRequest)input);
+        }
+
+        /// <summary>
+        /// Marshaller the request object to the HTTP request.
+        /// </summary>  
+        /// <param name="publicRequest"></param>
+        /// <returns></returns>
+        public IRequest Marshall(ListBucketMetricsConfigurationsRequest publicRequest)
+        {
+            var request = new DefaultRequest(publicRequest, "Amazon.S3");
+            PreMarshallCustomization(request, publicRequest);
+            request.HttpMethod = "GET";
+            request.AddSubResource("metrics");
+        
+            if (publicRequest.IsSetExpectedBucketOwner()) 
+            {
+                request.Headers["x-amz-expected-bucket-owner"] = publicRequest.ExpectedBucketOwner;
+            }
+            if (string.IsNullOrEmpty(publicRequest.BucketName))
+                throw new System.ArgumentException("BucketName is a required property and must be set before making this call.", "ListBucketMetricsConfigurationsRequest.BucketName");
+            
+            if (publicRequest.IsSetContinuationToken())
+                request.Parameters.Add("continuation-token", StringUtils.FromString(publicRequest.ContinuationToken));
```

Comment:
Similar to the other PR, we were doing sub resources before: https://github.com/aws/aws-sdk-net/blob/62c261a9ac72493e5a02cdcbf33b3a60a311ff48/sdk/src/Services/S3/Custom/Model/Internal/MarshallTransformations/ListBucketMetricsConfigurationsRequestMarshaller.cs#L51

Is that intentional here too?